lbz/authz/collector.py

```diff
@@ -0,0 +1,28 @@
+from lbz.misc import Singleton
+
+
+class AuthzCollector(metaclass=Singleton):
+    # TODO: Add uni tests
+    def __init__(
+        self,
+    ) -> None:
+        self.possible_permissions: list = []
+        self.resource = ""
+        self.guest_permissions: dict = {}
+
+    def set_resource(self, resource_name: str) -> None:
+        self.resource = resource_name
+
+    def set_guest_permissions(self, guest_permissions: dict) -> None:
+        self.guest_permissions = guest_permissions
+
+    def add_authz(self, permission: str) -> None:
+        self.possible_permissions.append(permission)
```

Comment:
In the near future, we will want to add more information about the permission. Maybe it is worth making it a dictionary right now?
```python
self.possible_permissions[permission] = None 
```
Currently `None` but in the future a description or more complicated structure. What do you think?